customers.go

```diff
@@ -169,6 +183,35 @@ func (c customerRepository) Update(customer shopify.Customer) (shopify.Customer,
 	return response.Customer.ToShopify(), nil
 }
 
+func (c customerRepository) GetByQuery(fields []string, query shopify.CustomerSearchQuery) (shopify.Customers, error) {
+	url := c.createURL(fmt.Sprintf("customers/search.json?fields=%v&query=%s", strings.Join(fields, ","), query.String()))
+
+	body, _, err := c.client.Get(url, nil)
+	if err != nil {
+		switch err.(http.ErrHTTP).Code {
+		case httpCode.StatusNotFound:
+			return shopify.Customers{}, shopify.NewErrCustomerSearchNotFound(query.String())
+		default:
+			return shopify.Customers{}, err
+		}
+	}
+
+	var responseDTO struct {
+		Customers CustomerDTOs `json:"customers"`
+	}
+
+	err = json.Unmarshal(body, &responseDTO)
+	if err != nil {
+		return shopify.Customers{}, err
+	}
+
+	if responseDTO.Customers[0].ID == 0 {
+		return shopify.Customers{}, shopify.NewErrCustomerSearchNotFound(query.String())
```

Comment:
I don't think this is an error, I think its perfectly find for a search to come back with 0 results :thinking:

Reply:
Yeh that's a good point probably don't need this.

Reply:
Yeah, maybe is not correct for the scope of this package. Perhaps this check should be done in FEMS?

I am saying that because we will always expect at least 1 result (1 customer) since each customer is linked to a patient. When a request is being sent over for deleting a customer/patient address to `patients/{pid}/addresses/{aid}`, it should always be possible to retrieve the customer from the patientID.

If that was not the case (no customer), then it will mean that we do not have a customer in Shopify for that patientID and that should be flagged since there is some sync issues between Shopify and PMS?

Reply:
So from a higher level and thinking about all the search queries/ searching in general, 0 results is not an error :). I think the user of the package can determine what to do with the results/ whether 0 should error but it shouldn't be a concern of this package.

Reply:
Will resolve as it looks like it has been removed anyway.